src/mono/mono/component/debugger-agent.c

```diff
@@ -5581,6 +5581,7 @@ decode_value_compute_size (MonoType *t, int type, MonoDomain *domain, guint8 *bu
 	if (type != t->type && !MONO_TYPE_IS_REFERENCE (t) &&
 		!(t->type == MONO_TYPE_I && type == MONO_TYPE_VALUETYPE) &&
 		!(type == VALUE_TYPE_ID_FIXED_ARRAY) &&
+		!(type == MDBGPROT_VALUE_TYPE_ID_NULL) &&
 		!(t->type == MONO_TYPE_U && type == MONO_TYPE_VALUETYPE) &&
 		!(t->type == MONO_TYPE_PTR && type == MONO_TYPE_I8) &&
 		!(t->type == MONO_TYPE_FNPTR && type == MONO_TYPE_I8) &&
```

Comment:
future cleanup: this should be a separate helper function. maybe `gboolean valuetype_is_blittable_assignable_from(MonoType *debugger_type, MonoType *expected_type)` ?  and make the condition `if (type != t->type && !MONO_TYPE_IS_REFERENCE(t) && !valuetype_is_blittable_assignable_from(type, t->type)) { ... error ... }`  (I'm not sure that is the best name)

Reply:
Thanks, @lambdageek , I will do it in my next PR on debugger-agent.